### PR TITLE
CORE-7071: Implement check for flow start permission

### PR DESCRIPTION
**Pull request by vkolomeyko**
Requires `corda-api` change: https://github.com/corda/corda-api/pull/650

### CODE REVIEW

File: components/flow/flow-rpcops-service-impl/src/main/kotlin/net/corda/flow/rpcops/impl/v1/FlowRPCOpsImpl.kt [hunk 1]
```diff
@@ -92,6 +100,14 @@ class FlowRPCOpsImpl @Activate constructor(
             throw InvalidInputDataException("The flow that was requested is not in the list of startable flows for this holding identity.")
         }
 
+        val rpcContext = CURRENT_RPC_CONTEXT.get()
+        val principal = rpcContext.principal
+
+        if (!permissionValidationService.permissionValidator.authorizeUser(principal,
```
Comment (vkolomeyko) [inline components/flow/flow-rpcops-service-impl/src/main/kotlin/net/corda/flow/rpcops/impl/v1/FlowRPCOpsImpl.kt:106]: Note: Follow-up Jira is scheduled: https://r3-cev.atlassian.net/browse/CORE-7180
which will allow passing vNodeId into authorization call.

**Review by ronanbrowne — APPROVED** [review]
LGTM (build)

**Review by benmcmahon100 — APPROVED** [review]

**Review by pmck91 — APPROVED** [review]

File: components/flow/flow-rpcops-service-impl/src/main/kotlin/net/corda/flow/rpcops/impl/v1/FlowRPCOpsImpl.kt [hunk 2]
```diff
@@ -92,6 +100,14 @@ class FlowRPCOpsImpl @Activate constructor(
...
+                "$START_FLOW_PREFIX$PREFIX_SEPARATOR${startFlow.flowClassName}")) {
```
Comment (sprintengine) [inline components/flow/flow-rpcops-service-impl/src/main/kotlin/net/corda/flow/rpcops/impl/v1/FlowRPCOpsImpl.kt:107]: So in effect, a user will need two permissions to run a flow:

```
StartFlow:com.cordapp.Calculator
POST:/api/v1/flow/{holdingIdentity}
```

File: components/flow/flow-rpcops-service-impl/src/main/kotlin/net/corda/flow/rpcops/impl/v1/FlowRPCOpsImpl.kt [hunk 3]
```diff
@@ -49,20 +55,22 @@ class FlowRPCOpsImpl @Activate constructor(
     @Reference(service = MessageFactory::class)
     private val messageFactory: MessageFactory,
     @Reference(service = CpiInfoReadService::class)
-    private val cpiInfoReadService: CpiInfoReadService
+    private val cpiInfoReadService: CpiInfoReadService,
+    @Reference(service = PermissionValidationService::class)
+    private val permissionValidationService: PermissionValidationService
```
Comment (sprintengine) [inline components/flow/flow-rpcops-service-impl/src/main/kotlin/net/corda/flow/rpcops/impl/v1/FlowRPCOpsImpl.kt:60]: I assume OSGi injects `PermissionValidationService` as a singleton? I.e. will injecting this in here mean the http/rpc worker has two PermissionValidationCacheServices - duplicating the cache? It looks like the default scope is singleton in `Component.scope()`.

File: components/flow/flow-rpcops-service-impl/src/main/kotlin/net/corda/flow/rpcops/impl/v1/FlowRPCOpsImpl.kt (hunk 2 above)
Comment (vkolomeyko) [reply components/flow/flow-rpcops-service-impl/src/main/kotlin/net/corda/flow/rpcops/impl/v1/FlowRPCOpsImpl.kt:107]: Yes, this is correct.

File: components/flow/flow-rpcops-service-impl/src/main/kotlin/net/corda/flow/rpcops/impl/v1/FlowRPCOpsImpl.kt (hunk 3 above)
Comment (vkolomeyko) [reply components/flow/flow-rpcops-service-impl/src/main/kotlin/net/corda/flow/rpcops/impl/v1/FlowRPCOpsImpl.kt:60]: Yeap, it is a singleton in the same JVM.

Comment (conalsmith-r3) [reply components/flow/flow-rpcops-service-impl/src/main/kotlin/net/corda/flow/rpcops/impl/v1/FlowRPCOpsImpl.kt:60]: Perfect!

**Pull request merged**